### PR TITLE
Refactor query builder sidebar 

### PR DESCRIPTION
**Pull request by gusaiani**
In preparation to add one more step to implement issue #17726 

The ViewSidebar component used a lib for the animation, probably due to a lack of familiarity with CSS. And adding new checks in the implementation file just to conditionally remove the animation would make the unclear code more confusing. It was trivial enough to rewrite.

Improves on issue https://github.com/metabase/metabase/issues/16704 as it makes the number of flickers go down to 1.

## How to Test

1. Ask a Question
2. Simple question
3. Sample dataset
4. Orders
5. Click on
  a. Visualization
  b. Settings
  c. Filter
  d. Summarize
  
 Sidebars should function just as before.

### CODE REVIEW

**Comment by gusaiani** [comment]
> Definitely safer to split this into two PRs, first to refactor and introduce a new version of the sidebar, and later to disable the animation. That way we get both the positive and negative tests of the reduced-motion behavior.

Ok, thanks. Done.

File: frontend/src/metabase/query_builder/components/view/ViewSidebar/ViewSidebar.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    right={right}
+    widthProp={width}
+    isOpen={isOpen}
+  >
+    <ViewSidebarContent widthProp={width}>{children}</ViewSidebarContent>
+  </ViewSidebarAside>
+);
+
+ViewSidebar.propTypes = {
+  left: PropTypes.bool,
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/view/ViewSidebar/ViewSidebar.jsx:21]: Based on the code it seems that `ViewSidebar` can be either `left` or `right`, could be improve props to reflect that? E.g. `side: PropTypes.oneOf(["left, "right"])`

File: frontend/src/metabase/query_builder/components/view/View.jsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ import QueryModals from "../QueryModals";
 import { ViewTitleHeader, ViewSubHeader } from "./ViewHeader";
 import NewQuestionHeader from "./NewQuestionHeader";
 import ViewFooter from "./ViewFooter";
-import ViewSidebar from "./ViewSidebar";
+import ViewSidebar from "./ViewSidebar/ViewSidebar";
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/view/View.jsx:27]: Could you add an `index.js` file with a re-export to remove nested import?

File: frontend/src/metabase/query_builder/components/view/ViewSidebar.jsx [hunk 3]
```diff
@@ -1,41 +0,0 @@
...
-import { Motion, spring } from "react-motion";
-
-const SPRING_CONFIG = { stiffness: 200, damping: 26 };
-
-const ViewSideBar = ({ left, right, width = 355, isOpen, children }) => (
-  <Motion
-    defaultStyle={{ opacity: 0, width: 0 }}
-    style={
-      isOpen
-        ? { opacity: spring(1), width: spring(width, SPRING_CONFIG) }
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/view/ViewSidebar.jsx:13]: @gusaiani Please also note that there would be difference in the way how it animates:
- previously `spring`
- now `linear`

**Review by ranquild — COMMENTED** [review]
I've checked everything and it seems to be working correctly, my only concerns are about changing animating timing function.

**Comment by gusaiani** [comment]
@mazameli,

in this refactor I changed the animation speed and easing.

Meant to make it snappier.

Can you take a look and tell us what you think, please?

**Review by ranquild — APPROVED** [review]
I'll approve the changes as code-wise they are OK, but please wait for @mazameli answer.

**Comment by gusaiani** [comment]
@ranquild Thanks for the review, I was waiting for green to let you know.
Your ideas made it better, no doubt about it o/

**Comment by gusaiani** [comment]
Merged as animation can be tweaked separately.

**Pull request merged**